pkg/controller/v1alpha1/llmisvc/scheduler.go

```diff
@@ -0,0 +1,481 @@
+
+	if llmSvc.Spec.Router != nil && llmSvc.Spec.Router.Scheduler != nil && llmSvc.Spec.Router.Scheduler.Template != nil {
+		podSpec := llmSvc.Spec.Router.Scheduler.Template.DeepCopy()
+
+		desiredPorts := sets.New("grpc", "grpc-health", "metrics")
+
+		actualPorts := make(map[string]*corev1.ContainerPort)
+		for _, container := range podSpec.Containers {
+			for _, port := range container.Ports {
+				if desiredPorts.Has(port.Name) {
+					actualPorts[port.Name] = port.DeepCopy()
+				}
+			}
+		}
+
+		if len(desiredPorts) != len(actualPorts) {
+			// TODO should this be raised as failing condition? + check if grpc port matches what's defined in the inferencepool
+			logger.Info("some ports are not matching", "desired", desiredPorts, "actual", maps.Keys(actualPorts))
+		}
+
+		var servicePorts []corev1.ServicePort
+		for name, port := range actualPorts {
+			servicePorts = append(servicePorts, corev1.ServicePort{
+				Name:       name,
+				Port:       port.ContainerPort,
+				TargetPort: intstr.FromString(name),
+				Protocol:   port.Protocol,
+			})
+		}
+
+		sort.Slice(servicePorts, func(i, j int) bool {
+			return servicePorts[i].Name < servicePorts[j].Name
+		})
+
+		svc.Spec.Ports = servicePorts
+	}
+
+	log.FromContext(ctx).V(2).Info("Expected router EPP service", "service", svc)
+
+	return svc
+}
+
+func (r *LLMISVCReconciler) expectedSchedulerInferencePool(ctx context.Context, llmSvc *v1alpha1.LLMInferenceService) *igwapi.InferencePool {
+	labels := r.schedulerLabels(llmSvc)
+
+	ip := &igwapi.InferencePool{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      kmeta.ChildName(llmSvc.GetName(), "-inference-pool"),
+			Namespace: llmSvc.GetNamespace(),
+			Labels:    labels,
+			OwnerReferences: []metav1.OwnerReference{
+				*metav1.NewControllerRef(llmSvc, v1alpha1.LLMInferenceServiceGVK),
+			},
+		},
+	}
+	if llmSvc.Spec.Router != nil && llmSvc.Spec.Router.Scheduler != nil && llmSvc.Spec.Router.Scheduler.Pool != nil && llmSvc.Spec.Router.Scheduler.Pool.Spec != nil {
+		ip.Spec = *llmSvc.Spec.Router.Scheduler.Pool.Spec.DeepCopy()
+	}
+
+	log.FromContext(ctx).V(2).Info("Expected router InferencePool", "inferencepool", ip)
+
+	return ip
+}
+
+func (r *LLMISVCReconciler) expectedSchedulerInferenceModel(ctx context.Context, llmSvc *v1alpha1.LLMInferenceService) *igwapi.InferenceModel {
+	labels := r.schedulerLabels(llmSvc)
+
+	im := &igwapi.InferenceModel{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      kmeta.ChildName(llmSvc.GetName(), "-inference-model"),
+			Namespace: llmSvc.GetNamespace(),
+			Labels:    labels,
+			OwnerReferences: []metav1.OwnerReference{
+				*metav1.NewControllerRef(llmSvc, v1alpha1.LLMInferenceServiceGVK),
+			},
+		},
+		Spec: igwapi.InferenceModelSpec{
+			ModelName: ptr.Deref(llmSvc.Spec.Model.Name, llmSvc.GetName()),
+			PoolRef: igwapi.PoolObjectReference{
+				Group: "inference.networking.x-k8s.io",
+				Kind:  "InferencePool",
+				Name:  igwapi.ObjectName(kmeta.ChildName(llmSvc.GetName(), "-inference-pool")),
```

Comment:
better to centralise naming pattern for all  names which are referenced in multiple functions. Same for -inference-model